packages/former/src/components/FormKitEdit.vue

```diff
@@ -45,25 +61,36 @@ const element = useAttrs() as FormKitSchemaNode;
 const mode = inject('mode');
 const schema = inject('schema');
 const selectedElementId = inject('selectedElementId');
+const {
+  reveal: openFormFieldSelector,
+  isRevealed: isFormFieldSelectorOpen,
+  confirm,
+  cancel: closeFormFieldSelector,
+} = useConfirmDialog<never, string, never>();
+const formFieldSelector = ref<HTMLElement | null>(null);
+onClickOutside(formFieldSelector, () => closeFormFieldSelector());
 
 const generateId = () => `former-${Math.random().toString(36).substring(7)}`;
 const getId = (element: any) => (element as { id?: string }).id;
 const id = computed(() => getId(element));
 
-function addComponentAfterThisOne() {
+async function addComponentAfterThisOne() {
+  const { data: elementType, isCanceled } = await openFormFieldSelector();
   if (!id.value) {
     throw new Error('This element should not have an add button');
   }
 
   // TODO: nested elements
   const index = schema.value.findIndex((e) => getId(e) === id.value);
-  schema.value.splice(index + 1, 0, {
-    $formkit: 'text',
-    id: generateId(),
-    name: 'new_field' + schema.value.length,
-    label: 'New field' + schema.value.length,
-    help: 'This is a new field.',
-  });
+  if (elementType && !isCanceled) {
```

Comment:
```suggestion
  if (!elementType || isCanceled) {
    return;
```

Would also make sense to do that directly after the call.